pkg/build/controller/common/util.go

```diff
@@ -134,3 +140,61 @@ func HasBuildPodNameAnnotation(build *buildapi.Build) bool {
 	_, hasAnnotation := build.Annotations[buildapi.BuildPodNameAnnotation]
 	return hasAnnotation
 }
+
+// ProcessValueFrom supports the k8s pod syntax of allowing "valueFrom" in addition
+// to "value", where "valueFrom" uses the downward api to reference existing env
+// variables, jsonpath references to the build object, secrets, and configmaps.
+func ProcessValueFrom(build *buildapi.Build, client kclientset.Interface) (*[]kapi.EnvVar, error) {
```

Comment:
(comment applies to whole PR) I think that working with types of *[]Foo is generally an antipattern.  The implementation of []Foo is that it already contains a pointer to an underlying shared array.  For this PR I'd sooner see rework to use []kapi.EnvVar; perhaps this will necessitate the addition of a helper function such as PutEnvToStrategy (or something), but I think that's preferable to *[]kapi.EnvVar.